src/main/webapp/app/exercises/shared/exercise-scores/exercise-scores-export-button.component.ts

```diff
@@ -3,68 +3,127 @@ import { roundValueSpecifiedByCourseSettings } from 'app/shared/util/utils';
+    private constructCSV(exercise: Exercise, withTestCases?: boolean, withFeedback?: boolean) {
         this.resultService.getResultsWithPointsPerGradingCriterion(exercise).subscribe((data) => {
             const results: ResultWithPointsPerGradingCriterion[] = data.body || [];
             if (results.length === 0) {
-                this.alertService.warning(`artemisApp.exercise.exportResultsEmptyError`, { exercise: exercise.title });
+                this.alertService.warning(`artemisApp.exercise.exportResults.emptyError`, { exercise: exercise.title });
                 window.scroll(0, 0);
                 return;
             }
             const isTeamExercise = !!(results[0].result.participation! as StudentParticipation).team;
             const gradingCriteria: GradingCriterion[] = ExerciseScoresExportButtonComponent.sortedGradingCriteria(exercise);
 
-            const keys = ExerciseScoresRowBuilder.keys(exercise, isTeamExercise, gradingCriteria);
-            const rows = results.map((resultWithPoints) => {
-                const studentParticipation = resultWithPoints.result.participation! as StudentParticipation;
-                return new ExerciseScoresRowBuilder(exercise, gradingCriteria, studentParticipation, resultWithPoints).build();
-            });
-
+            let keys;
+            let rows;
+            if (withTestCases) {
+                const testCasesNames = this.getTestCaseNamesFromResults(results);
+                keys = ExerciseScoresRowBuilder.keys(exercise, isTeamExercise, gradingCriteria, testCasesNames);
+                rows = results.map((resultWithPoints) => {
+                    const studentParticipation = resultWithPoints.result.participation! as StudentParticipation;
+                    const testCaseResults = this.getTestCaseResults(resultWithPoints, testCasesNames, withFeedback);
+                    return new ExerciseScoresRowBuilder(exercise, gradingCriteria, studentParticipation, resultWithPoints, testCaseResults).build();
+                });
+            } else {
+                keys = ExerciseScoresRowBuilder.keys(exercise, isTeamExercise, gradingCriteria);
+                rows = results.map((resultWithPoints) => {
+                    const studentParticipation = resultWithPoints.result.participation! as StudentParticipation;
+                    return new ExerciseScoresRowBuilder(exercise, gradingCriteria, studentParticipation, resultWithPoints).build();
+                });
+            }
             const fileNamePrefix = exercise.shortName ?? exercise.title?.split(/\s+/).join('_');
-            ExerciseScoresExportButtonComponent.exportAsCsv(`${fileNamePrefix}-results-scores`, keys, rows);
+
+            if (withFeedback) {
+                ExerciseScoresExportButtonComponent.exportAsCsv(`${fileNamePrefix}-results-scores`, keys, rows, ',');
+            } else {
+                ExerciseScoresExportButtonComponent.exportAsCsv(`${fileNamePrefix}-results-scores`, keys, rows);
```

Comment:
Why do we need different separators here?

Reply:
Using the currently used `;` separator seems not to work with feedbacks containing newlines, but did work with the more standard comma-separator.. As I didn't want to change the current functionality but incrementally add this feature, I switch between the two separators here